### PR TITLE
Huge code cleanup [v5]

### LINKED ISSUE #841: Mistakes in ModelAbstractExpandableItem class
`ModelAbstractExpandableItem` contains problems:
 - unused parameter `Parent`
   - can be used in `var parent: IParentItem<*>?` as wildcard type (redundant clarification IMHO)
   - can be removed as redundant
- unused parameter `SubItems`
   - can be used in `var subItems: MutableList<ISubItem<*>>` as wildcard type (redundant clarification IMHO)
   - can be removed as redundant
 - wrong behavior of `isSelectable` field
   - getter must be changed from `_subItems.isNotEmpty()` to `_subItems.isEmpty()` because expandable item must be selectable only if not contains sub-items
 - setter of `subItems` duplicate logic from `MutableSubItemList`
   - `_subItems` can be changed to `val`, clear and add new items from setter
   - `subItems` can be declared as `MutableSubItemList<ISubItem<*>>` and as `val`. Also add method for clear and fill from another collection to `MutableSubItemList` (best way IMHO)

Solutions of this problems makes some breaking changes and can be provided in 5.x version. I'll make PR with selected by you ways

### PR DESCRIPTION
**Pull request by zTrap**
- replaced usage of 'magic number' `-1` by `RecyclerView.NO_POSITION` constant
 - added `@JvmStatic` for all functions in `Util object classes`
 - cleared some explicit parameters
 - cleared documentations
 - added function for removing `PagedListListener<Model>` from `PagedItemListImpl`
 - bumped versions of dependencies
 - optimized some methods and classes

closes #841

### CODE REVIEW

**Comment by zTrap** [comment]
Sorry for a long pause. I was high loaded at work

File: library-core/src/main/java/com/mikepenz/fastadapter/MutableSubItemList.kt [hunk 1]
```diff
@@ -43,4 +43,14 @@ class MutableSubItemList<E : ISubItem<*>>(val parent: IParentItem<*>, val list:
         element.parent = parent
         return list.set(index, element).also { oldElement -> oldElement.parent = null }
     }
+
+    override fun clear() {
+        list.forEach { it.parent = null }
+        list.clear()
+    }
+
+    fun setNewList(newList: List<E>) {
```
Comment (mikepenz) [inline library-core/src/main/java/com/mikepenz/fastadapter/MutableSubItemList.kt]: set new list should not clear and add, but instead actually cause a complete new list reference

**Comment by mikepenz** [comment]
There seems an issue with travis though: https://travis-ci.org/mikepenz/FastAdapter/builds/628764909?utm_source=github_status&utm_medium=notification

**Comment by mikepenz** [comment]
And don't worry for the pause. The changes look good. I like that we finally got rid of the `m*` and the `-1` :D

File: library-core/src/main/java/com/mikepenz/fastadapter/MutableSubItemList.kt (hunk 1 above)
Comment (zTrap) [reply library-core/src/main/java/com/mikepenz/fastadapter/MutableSubItemList.kt]: changed to re-link to reference of new list

**Comment by zTrap** [comment]
Actually `app` also need to cleanup (contains some wrong docs, `m*` and magic numbers), but I thought PR already huge for this changes :D I think I'll cleanup it too, but later

**Comment by mikepenz** [comment]
```
> Task :app:compileReleaseKotlin FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':app:compileReleaseKotlin'.
> Internal compiler error. See log for more details
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
```

travis seems sad :D

**Comment by zTrap** [comment]
Try to re-run, maybe it's temporary error `e: java.lang.OutOfMemoryError: Java heap space`

**Comment by mikepenz** [comment]
seems to be persistent. let's see if we can somehow adjust the memory allowance for travis

**Pull request merged**